k9mail/src/main/java/com/fsck/k9/view/MessageWebView.java

```diff
@@ -102,19 +104,18 @@ private void disableDisplayZoomControls() {
         getSettings().setDisplayZoomControls(!supportsMultiTouch);
     }
 
-    /**
-     * Load a message body into a {@code MessageWebView}
-     *
-     * <p>
-     * Before loading, the text is wrapped in an HTML header and footer
-     * so that it displays properly.
-     * </p>
-     *
-     * @param text
-     *      The message body to display.  Assumed to be MIME type text/html.
-     */
-    public void setText(String text) {
-     // Include a meta tag so the WebView will not use a fixed viewport width of 980 px
+    public void displayHtmlContentWithInlineAttachments(String htmlText, AttachmentResolver attachmentResolver) {
+        setAttachmentResolverWebViewClient(attachmentResolver);
+        setHtmlContent(htmlText);
+    }
+
+    private void setAttachmentResolverWebViewClient(AttachmentResolver attachmentResolver) {
+        K9WebViewClient webViewClient = K9WebViewClient.newInstance(attachmentResolver);
```

Comment:
Looks like `K9WebViewClient` might throw a `NullPointerException` when the `attachmentResolver` passed to this method is `null`.